packages/firebase_auth/firebase_auth/example/test_driver/instance_e2e.dart

```diff
@@ -123,6 +123,34 @@ void runInstanceTests() {
         await subscription.cancel();
       });
 
+      test('fires once on first initialization of FirebaseAuth', () async {
+        /* fixes a very specific bug: https://github.com/FirebaseExtended/flutterfire/issues/3628
+        If the first initialization of FirebaseAuth involves the listeners userChanges() or idTokenChanges()
+        the user will receive two events. Why? The native SDK listener will always fire an event upon initial
+        listen. FirebaseAuth also sends an initial synthetic event. We send a synthetic event because, ordinarily, the user will
+        not use a listener as the first occurrence of FirebaseAuth. We, therefore, mimic native behaviour by sending an
+        event. This test proves the logic of PR: https://github.com/FirebaseExtended/flutterfire/pull/6560
+         */
+
+        //requires a fresh app
+        FirebaseApp second = await Firebase.initializeApp(
+            name: 'test-init',
+            options: const FirebaseOptions(
+              apiKey: 'AIzaSyAHAsf51D0A407EklG1bs-5wA7EbyfNFg0',
+              appId: '1:448618578101:ios:4cd06f56e36384acac3efc',
+              messagingSenderId: '448618578101',
+              projectId: 'react-native-firebase-testing',
+              authDomain: 'react-native-firebase-testing.firebaseapp.com',
+              iosClientId:
+                  '448618578101-m53gtqfnqipj12pts10590l37npccd2r.apps.googleusercontent.com',
+            ));
+        Stream<User> stream =
+            FirebaseAuth.instanceFor(app: second).userChanges();
+
+        subscription = stream.listen(expectAsync1((event) {},
+            count: 1, reason: 'Stream should only call once'));
```

Comment:
I could see this test pass even if we send two events, because the test could complete before the second event is emitted

Reply:
Maybe an alternative would be to trigger a user change, and test that we received 2 events instead of 3?

like:

```dart
await expectLater(stream, emitsInOrder(initial event, new user name));

change user name
```

Reply:
Yeah, I tried that but I wasn't able to get it to work. Also - The user was being persisted so each test run had to be conditional on whether the user was signed in/signed out from the previous run which just made it messy. I think the purpose of the test is to prove the number of events is correct, not the event itself. Using a simple `Future.delayed` passed/failed correctly every time I added/removed this [logic](https://github.com/FirebaseExtended/flutterfire/pull/6560/files#diff-4173fb0903a19bd9e482c2e8c78b8077fee1c144a6ab6c66121bd039259a79c3R26-R29).